pkg/servicemesh/federation/server/server.go

```diff
@@ -234,68 +230,23 @@ func (s *Server) Run(stopCh <-chan struct{}) {
 	_ = s.httpServer.Shutdown(context.TODO())
 }
 
-func (s *Server) GetGatewayEndpoints() []*federationmodel.ServiceEndpoint {
-	s.Lock()
-	defer s.Unlock()
-	return append([]*federationmodel.ServiceEndpoint(nil), s.currentGatewayEndpoints...)
-}
-
-func (s *Server) resyncNetworkGateways() (bool, error) {
-	s.Lock()
-	defer s.Unlock()
-
-	gatewayEndpoints := []*federationmodel.ServiceEndpoint{}
-	for _, gateway := range s.env.NetworkGateways() {
-		gatewayEndpoints = append(gatewayEndpoints, &federationmodel.ServiceEndpoint{
-			Port:     int(gateway.Port),
-			Hostname: gateway.Addr,
-		})
-	}
-
-	newGatewayChecksum, err := hashstructure.Hash(gatewayEndpoints, hashstructure.FormatV2, &hashstructure.HashOptions{SlicesAsSets: true})
-	if err != nil {
-		return false, err
-	}
-
-	oldGatewayChecksum, err := hashstructure.Hash(s.currentGatewayEndpoints, hashstructure.FormatV2, &hashstructure.HashOptions{SlicesAsSets: true})
-	if err != nil {
-		return false, err
-	}
-	if oldGatewayChecksum != newGatewayChecksum {
-		s.currentGatewayEndpoints = gatewayEndpoints
-		return true, nil
-	}
-	return false, nil
-}
-
 func (s *Server) UpdateService(svc *model.Service, event model.Event) {
 	// this might be a NetworkGateway
 	if svc != nil {
```

Comment:
this whole if-block needs to be removed. this was for sending an empty update event in case a gateway was updated - we're not sending gateways anymore, so this can go. That will then need a bunch of changes in the tests again, especially TestWatch